### PR TITLE
Object Calisthenics - best practise

### PR DESCRIPTION
**Pull request by alikon**
as per discussion on https://github.com/joomla/joomla-cms/pull/20417#issuecomment-390393952

### CODE REVIEW

File: manual/basic-guidelines.md [hunk 1]
```diff
@@ -29,3 +29,19 @@ There is no maximum limit for line lengths in files, however, a notional value o
...
+    2. Don’t Use The ELSE Keyword
+    3. Wrap All Primitives And Strings
+    4. First Class Collections
+    5. One Dot Per Line
+    6. Don’t Abbreviate
+    7. Keep All Entities Small
+    8. No Classes With More Than Two Instance Variables
+    9. No Getters/Setters/Properties
+
+for more info:
```
Comment (brianteeman) [inline manual/basic-guidelines.md]: For more info:

**Comment by PhilETaylor** [comment]
And who is going to apply these new nine things to the existing code??? 

No point in new standards when 90% (made up number) of the existing code would then be non compliant!

**Comment by brianteeman** [comment]
always lookingg to find faults :(

**Comment by PhilETaylor** [comment]
Copy and pasting random opinions (these being Jeffs opinions) is not the way to design best practice for a mass marked code base

**Comment by alikon** [comment]
thank you for your words, these words really encourage

**Comment by PhilETaylor** [comment]
I think it’s absurd that a self described non coder believes he has the mandate to dictate code standards - and also to repeatedly abuse contributors

The 9 rules are just opinion. There are equally others that disagree with these coding opinions. There always will be. But as a project a discussion should be had and a standard adopted. Not just approved in seconds by a non coder. 

If only had a community policy with teeth - there would be plenty of evidence to banish Teeman for good. The personal attacks are relentless and well documented. 

——
Sent from my iPhone - sorry - if needed I’ll send longer email from my desk later.
——

**Comment by alikon** [comment]
nemesis

**Comment by PhilETaylor** [comment]
Brian you are toxic. Period

Fact - I ASKED FOR THIS PR. 
fact this PR CAME FROM A PR OF MY CODE!

Therefore your argument is invalid. As always.

Once again I’m blocking you. Life is too short to put up with personal attacks. 


——
Sent from my iPhone - sorry - if needed I’ll send longer email from my desk later.
——

**Comment by PhilETaylor** [comment]
In fact Teeman is PERSONALLY RESPONSIBLE for loss of people like me contributing less.

——
Sent from my iPhone - sorry - if needed I’ll send longer email from my desk later.
——

**Comment by alikon** [comment]
reopened, just the time to dress an anti-gas mask

**Comment by mbabker** [comment]
Object Calisthenics is heavily opinionated and unenforceable in the Joomla code without extremely disruptive B/C breaks.  So no, this can't just be blindly copy/pasted in and enforced as part of the coding standard.

**Pull request closed without merging**